common/softwarecontainer-common.cpp

```diff
@@ -75,22 +77,38 @@ bool isDirectoryEmpty(const std::string &path) {
 
 bool isFile(const std::string &path)
 {
-    return S_ISREG(getStat(path).st_mode);
+    struct stat st;
+    if (getStat(path, st)) {
+        return S_ISREG(st.st_mode);
+    }
+
+    return false;
 }
 
 bool isPipe(const std::string &path)
 {
-    return S_ISFIFO(getStat(path).st_mode);
+    struct stat st;
+    if (getStat(path, st)) {
+        return S_ISFIFO(st.st_mode);
+    }
+
+    return false;
 }
 
 bool isSocket(const std::string &path)
 {
-    return S_ISSOCK(getStat(path).st_mode);
+    struct stat st;
+    if (getStat(path, st)) {
+        return S_ISSOCK(st.st_mode);
+    }
+
+    return false;
 }
 
 bool existsInFileSystem(const std::string &path)
```

Comment:
This seems like a weird call overall.. Why would you ever check this? What you want is to stat if the file exists and is in some way accessible, right?